config.m4

```diff
@@ -309,12 +320,19 @@ if test "$PHP_MONGODB" != "no"; then
     dnl Generated with: find src/libmongoc/src/zlib-1.2.11 -maxdepth 1 -name '*.c' -print0 | cut -sz -d / -f 5- | sort -z | tr '\000' ' '
     PHP_MONGODB_ZLIB_SOURCES="adler32.c compress.c crc32.c deflate.c gzclose.c gzlib.c gzread.c gzwrite.c infback.c inffast.c inflate.c inftrees.c trees.c uncompr.c zutil.c"
 
-    PHP_ADD_SOURCES_X(PHP_EXT_DIR(mongodb)[src/libmongoc/src/mongoc], $PHP_MONGODB_MONGOC_SOURCES, $PHP_MONGODB_MONGOC_CFLAGS, shared_objects_mongodb, yes)
+    if test "$ext_shared" = "no"; then
+      PHP_ADD_SOURCES(PHP_EXT_DIR(mongodb)[/src/libmongoc/src/mongoc], $PHP_MONGODB_MONGOC_SOURCES, $PHP_MONGODB_MONGOC_CFLAGS)
+    else
+      PHP_ADD_SOURCES_X(PHP_EXT_DIR(mongodb)[src/libmongoc/src/mongoc], $PHP_MONGODB_MONGOC_SOURCES, $PHP_MONGODB_MONGOC_CFLAGS, shared_objects_mongodb, yes)
+    fi
 
-    m4_include(scripts/build/autotools/m4/pkg.m4)
+    sinclude(ext/mongodb/scripts/build/autotools/m4/pkg.m4)
+    sinclude(ext/mongodb/scripts/build/autotools/CheckHost.m4)
+    sinclude(ext/mongodb/scripts/build/autotools/CheckSSL.m4)
 
-    m4_include(scripts/build/autotools/CheckHost.m4)
-    m4_include(scripts/build/autotools/CheckSSL.m4)
+    sinclude(scripts/build/autotools/m4/pkg.m4)
+    sinclude(scripts/build/autotools/CheckHost.m4)
+    sinclude(scripts/build/autotools/CheckSSL.m4)
```

Comment:
``sinclude`` works like ``m4_include``, but doesn't bail out if it can't find the file. As this happens during ``./buildconf``, we can't make use of shell magic so this seems the only (ugly) way: we just try them both.

Reply:
How is this different than deciding to call `PHP_ADD_SOURCES()` or `PHP_ADD_SOURCES_X()` based on the value of `$ext_shared`? Does that solution not apply here because `m4_include()` is evaluated earlier to generate the `configure` script, long before we'd be able to check the value of `$ext_shared`?

I tried using `m4_include(PHP_EXT_DIR(mongodb)[scripts/build/autotools/m4/pkg.m4])` but that results in:

```
/usr/bin/m4:config.m4:314: cannot open `""scripts/build/autotools/m4/pkg.m4': No such file or directory
```

Is `PHP_EXT_DIR(mongodb)` only an empty string when building a shared extension? I found it [declared in `phpize.m4`](https://github.com/php/php-src/blob/39c58571350692b4cee3203280546f795335f19e/scripts/phpize.m4#L11) as an empty string. There is a more substantial definition in [`configure.ac`](https://github.com/php/php-src/blob/master/configure.ac#L123), but I assume that's for built-in extensions.